content/blog/2022-06-29-june-community-gems.md

```diff
@@ -0,0 +1,200 @@
+$ dvc exp show --drop 'Created|train.seed|./clf|./data/*|./src/train.py|src/evaluate.py'
+```
+
+Then get a table like this:
+
+```dvctable
+ ─────────────────────────────────────────────────────────────────
+  neutral:**Experiment**   metric:**avg_prec**   metric:**roc_auc**   param:**train.n_est**   param:**train.min_split**
+ ─────────────────────────────────────────────────────────────────
+  workspace           -         -   300           75
+  mlem-serve    0.76681   0.38867   300           75
+ ─────────────────────────────────────────────────────────────────
+```
+
+If you're interested in more advanced visualizations, you should try out
+[Iterative Studio](https://studio.iterative.ai/#features).
+
+## [Is it possible to create, commit, and push updates to datasets using DVC with Python instead of the command line?](https://discord.com/channels/485586884165107732/563406153334128681/988895726257991740)
+
+Fantastic question from @wlu07!
+
+Yes, we do have an internal `Repo` class to do DVC operations using Python. You
+can refer to the
+[GitHub repo for the DVC CLI commands](https://github.com/iterative/dvc/tree/main/dvc/commands)
+to see how the CLI arguments are translated into the `Repo` function arguments
+and you can see how to use some of the
+[`Repo` methods in our docs](https://dvc.org/doc/api-reference).
+
+Here's an example of how you might run DVC commands using Python:
+
+```python
+from dvc.repo import Repo
+
+repo = Repo(".")
+
+repo.add("test_dataset.csv")
+
+repo.push()
+```
+
+Keep in mind that `dvc.repo.Repo` is not an official public API, so there is no
+guarantee it will always be in stable state.
+
+## [How can I write generated artifacts back to a GitHub repo after a GitHub workflow is finished?](https://discord.com/channels/485586884165107732/728693131557732403/983379949023006750)
+
+Wonderful CML question from @Fourtin!
+
+If you want to add the artifact to your repo just like you would a file, then
+you should check out the [`cml pr <file>` command](https://cml.dev/doc/ref/pr).
+You can use this to merge pull requests to the same branch the workflow was
+triggered from.
+
+For example, if you run a command like:
+
+```dvc
+$ cml pr --squash train.py
+```
+
+It will run `git add train.py`, commit the change, create a new branch, open a
+pull request, and squash and merge it.
+
+## [Is there a way to programmatically update the content of `params.py`?](https://discord.com/channels/485586884165107732/563406153334128681/987004036995764304)
+
+Thanks for asking this @petek!
+
+If you have a `params.py` file like this:
+
+```python
+class TrainTestSplit:
+    FOLDER = "data/train_test_split"
+    SPLIT_METHOD = "proportional"
+```
+
+In DVC, you can update the params and run `dvc exp run --set-param <param>`.
+Here's an example of what that might look like:
+
+```dvc
+$ dvc exp run --set-param params.py:TrainTestSplit.SPLIT_METHOD="proportional"
+```
+
+_Note:_ It may not be able to update Python parameters correctly.
```

Comment:
Depending on what? Are we unsure whether this works at all, or is DVC flaky in this instance? 😅
Maybe we could provide a nudge towards using a `params.yaml`? Or is `params.py` a common approach?

Reply:
> Depending on what?

From https://dvc.org/doc/command-reference/params#examples-python-parameters-file :

> ⚠️ Note that complex expressions (unsupported by [ast.literal_eval](https://docs.python.org/3/library/ast.html#ast.literal_eval)) won't be parsed as DVC parameters.

>  Maybe we could provide a nudge towards using a params.yaml? Or is params.py a common approach?

We already nudge towards `params.yaml` by making it the default. 
Outside of that, I think we should let users choose any of the supported formats and explicitly state limitations.